cmake/kokkos_test_cxx_std.cmake

```diff
@@ -86,6 +86,17 @@ ELSE()
   MESSAGE(FATAL_ERROR "Unknown C++ standard ${KOKKOS_CXX_STANDARD} - must be 14, 17, or 20")
 ENDIF()
 
+# Enforce that we can compile a simple C++14 program
+
+TRY_COMPILE(CAN_COMPILE_CPP14
+  ${KOKKOS_TOP_BUILD_DIR}/corner_cases
+  ${KOKKOS_SOURCE_DIR}/cmake/compile_tests/cplusplus14.cpp
+)
+if (NOT CAN_COMPILE_CPP14)
+  UNSET(CAN_COMPILE_CPP14 CACHE) #make sure CMake always re-runs this
+  MESSAGE(FATAL_ERROR "C++${KOKKOS_CXX_STANDARD} specified, but cannot compile a minimal C++14 or later program. Perhaps the compiler is not set up correctly (e.g., platform standard headers are being used instead of compiler standard headers)?")
```

Comment:
This is probably fine. Sorry, I should have been clearer. I was expecting something more like: 

"C++{...}-compliant compiler detected, but unable to compile C++14 or later program. Verify that ${CMAKE_CXX_COMPILER_ID}:${CMAKE_CXX_COMPILER_VERSION} is set up correctly (e.g. check that correct library headers are being used)."

Reply:
Updated the error message to match this one.